sprint_15

anton-kashnikoff opened the pull request.

Commit:
Added new irregular event view controller

Commit:
Renamed some variables

Commit:
Fixed isDataReadyForTracker function

Commit:
Added CoreData to save trackers

Commit:
Created custom table view cells to show schedule table view

Commit:
Created folder for table view cells

Commit:
Reorganized files and folders

Commit:
Added static properties to class UIImage

So1ovyov commented on Tracker/AppDelegate.swift:
```diff
@@ -6,6 +6,7 @@
 //
 
 import UIKit
+import CoreData
```
Было бы круто расставить импорты в алфавитном порядке.

So1ovyov commented on Tracker/CoreData/TrackerCategoryStore.swift:
```diff
@@ -0,0 +1,109 @@
+//
+
+import UIKit
+import CoreData
```
Было бы круто расставить импорты в алфавитном порядке.

So1ovyov commented on Tracker/CoreData/TrackerCategoryStore.swift:
```diff
@@ -0,0 +1,109 @@
+        }
+    }
+    
+//    func getTrackerCategories() -> [TrackerCategory] {
```
Закомментированный код можно удалить, если он не используется.

So1ovyov commented on Tracker/CoreData/TrackerCategoryStore.swift:
```diff
@@ -0,0 +1,109 @@
+        fetchedResultsController.fetchedObjects == nil
+    }
+    
+//    func getAllObjects() -> [TrackerCategoryCoreData] {
```
Закомментированный код можно удалить, если он не используется.

So1ovyov commented on Tracker/CoreData/TrackerRecordStore.swift:
```diff
@@ -0,0 +1,87 @@
+//
+
+import UIKit
+import CoreData
```
Было бы круто расставить импорты в алфавитном порядке.

So1ovyov commented on Tracker/CoreData/TrackerStore.swift:
```diff
@@ -0,0 +1,111 @@
+//
+
+import UIKit
+import CoreData
```
Было бы круто расставить импорты в алфавитном порядке.

So1ovyov commented on Tracker/CoreData/TrackerStore.swift:
```diff
@@ -0,0 +1,111 @@
+        }
+    }
+    
+//    func getTrackers(of categoryObject: TrackerCategoryCoreData) -> [Tracker] {
```
Закомментированный код можно удалить, если он не используется.

So1ovyov commented on Tracker/CoreData/TrackerStore.swift:
```diff
@@ -0,0 +1,111 @@
+        fetchedResultsController.sections?[section].numberOfObjects ?? 0
+    }
+    
+//    func numberOfObjects() -> Int {
```
Закомментированный код можно удалить, если он не используется.

So1ovyov commented on Tracker/CoreData/TrackerStore.swift:
```diff
@@ -0,0 +1,111 @@
+        fetchedResultsController.fetchedObjects == nil || fetchedResultsController.fetchedObjects == []
+    }
+    
+//    func getAllObjects() -> [TrackerCoreData] {
```
Закомментированный код можно удалить, если он не используется.

So1ovyov commented on Tracker/Helpers/DataHelper.swift:
```diff
@@ -6,96 +6,73 @@
-        }
-    }
+//
+//final class DataHelper {
```
Закомментированный код можно удалить, если он не используется.

So1ovyov replied:
Здесь и ниже во всем файле.

So1ovyov commented on Tracker/Models/Tracker.swift:
```diff
@@ -7,7 +7,15 @@
 import UIKit
 
-struct Tracker {
+protocol TrackerProtocol {
```
А точно ли нужны протоколы, когда можно создать просто модельку структуру.

So1ovyov commented on Tracker/ViewControllers/TrackersViewController.swift:
```diff
@@ -209,77 +191,53 @@ final class TrackersViewController: UIViewController {
-            imageView.image = UIImage(named: "star")
-            label.text = "Что будем отслеживать?"
+        if trackerStore.isFetchedObjectsEmpty() && currentText != "" {
+            print("FIRST")
```
Принты можно удалить, если они только для дебага.

Commit:
Deleted commented code
Imports are arranged in alphabetical order
Removed unused files
Deleted unused print-functions
